shared/actions/login.js

```diff
@@ -152,6 +152,11 @@ function* navBasedOnLoginAndInitialState(): Saga.SagaGenerator<any, any> {
     yield Saga.put(switchRouteDef(loginRouteTree))
     yield Saga.put.resolve(getExtendedStatus())
     yield Saga.call(getAccounts)
+    // We may have logged successfully in by now, check before trying to navigate
+    const state = yield Saga.select()
+    if (state.config.loggedIn) {
+      return
+    }
```

Comment:
Perhaps `switchRouteDef` should take an initial route to navigate to, to avoid similar races. WDYT?

Reply:
I'm not sure how that would fix the problem, what do you mean?

Reply:
Ah, never mind, I guess this is a separate problem.